Emby.Server.Implementations/ScheduledTasks/Tasks/AudioNormalizationPostScanTask.cs

```diff
@@ -0,0 +1,198 @@
+        foreach (var library in _libraryManager.RootFolder.Children)
+        {
+            var libraryOptions = _libraryManager.GetLibraryOptions(library);
+            if (!libraryOptions.EnableLUFSScan)
+            {
+                continue;
+            }
+
+            // Album gain
+            var albums = _libraryManager.GetItemList(new InternalItemsQuery
+            {
+                IncludeItemTypes = [BaseItemKind.MusicAlbum],
+                Parent = library,
+                Recursive = true
+            });
+
+            foreach (var a in albums)
+            {
+                if (a.NormalizationGain.HasValue || a.LUFS.HasValue)
+                {
+                    continue;
+                }
+
+                // Skip albums that don't have multiple tracks, album gain is useless here
+                var albumTracks = ((MusicAlbum)a).Tracks.Where(x => x.IsFileProtocol).ToList();
+                if (albumTracks.Count <= 1)
+                {
+                    continue;
+                }
+
+                var tempFile = Path.Join(_configurationManager.GetTranscodePath(), Guid.NewGuid() + ".concat");
+                var inputLines = albumTracks.Select(x => string.Format(CultureInfo.InvariantCulture, "file '{0}'", x.Path.Replace("'", @"'\''", StringComparison.Ordinal)));
+                await File.WriteAllLinesAsync(tempFile, inputLines, cancellationToken).ConfigureAwait(false);
+                a.LUFS = await CalculateLUFSAsync(
+                    string.Format(CultureInfo.InvariantCulture, "-f concat -safe 0 -i \"{0}\"", tempFile),
+                    cancellationToken).ConfigureAwait(false);
+                File.Delete(tempFile);
+            }
+
+            _itemRepository.SaveItems(albums, cancellationToken);
+
+            // Track gain
+            var tracks = _libraryManager.GetItemList(new InternalItemsQuery
+            {
+                MediaTypes = [MediaType.Audio],
+                IncludeItemTypes = [BaseItemKind.Audio],
+                Parent = library,
+                Recursive = true
+            });
+
+            foreach (var t in tracks)
+            {
+                if (t.NormalizationGain.HasValue || t.LUFS.HasValue || !t.IsFileProtocol)
+                {
+                    continue;
+                }
+
+                t.LUFS = await CalculateLUFSAsync(string.Format(CultureInfo.InvariantCulture, "-i \"{0}\"", t.Path.Replace("\"", "\\\"", StringComparison.Ordinal)), cancellationToken);
+            }
+
+            _itemRepository.SaveItems(tracks, cancellationToken);
+        }
+    }
+
+    /// <inheritdoc />
+    public IEnumerable<TaskTriggerInfo> GetDefaultTriggers()
+    {
+        return
+        [
+            new TaskTriggerInfo
+            {
+                Type = TaskTriggerInfo.TriggerInterval,
+                IntervalTicks = TimeSpan.FromHours(24).Ticks
+            }
+        ];
+    }
+
+    private string EscapeFilename(string filename)
+        => filename;
```

Comment:
Why have this function?

Reply:
Believe it or not, but it actually used to do something :p
But since escaping for albums and tracks is different this isn't used anymore